test/binCases/errors/plugin-options-validation/test.js

```diff
@@ -0,0 +1,13 @@
+"use strict";
+
+module.exports = function testAssertions(code, stdout, stderr) {
+	code.should.be.eql(1);
+	stdout.should.be.empty();
+
+	stderr[1].should.containEql("Commons Chunk Plugin Invalid Options");
+
+	stderr[3].should.containEql("options should be string");
```

Comment:
Did you mean this messages by 'not good ' ? 😛 These are `defaults` ? Do we need them ? Based on the `webpack.config.js` I only expected `"options['foo'] should NOT have additional properties"` here, but that's up to sokra to decide

Reply:
This and the actual error messaging format is not very good. In this case, `oneOf` is the reason for these error messages: CommonsChunkPlugin options could be a string, array or object and AJV reports errors on every `oneOf` as if all of them failed. 

This happens in the webpackOptionValidation too, but the [error messaging there is changed](https://github.com/webpack/webpack/blob/master/lib/WebpackOptionsValidationError.js#L144) in a way, that will print this error a lit more nicely. It will look something like this:

```
options has an unknown property 'foo'

options should be one of these:
* string
* array[string]
* { <object schema> }
```

Reply:
We can definitely handle `anyOf, oneOf, default` etc in a better way with better messages, `schema-utils` does the minimum atm. If you have something in mind feel free to open an issue or PR if you have the time.